### PR TITLE
Newpct provider fix

### PR DESCRIPTION
**Pull request by CristianBB**
Adapted to new version

### CODE REVIEW

**Comment by SiCKRAGEB0T** [comment]
Thanks for the pull request! Before a real human comes by, please make sure your report has all the below criteria checked

- [x] Give a description on what the PR is for.
- [x] Make sure you send a PR against the DEVELOP branch
- [x] Don't send big changes in one go. Split up big PRs into multiple smaller (easier manageble) PRs

Please make sure you also read [contribution guide](https://github.com/SiCKRAGETV/SickRage/blob/master/contributing.md) and followed all the steps.

Thanks!

File: sickbeard/providers/newpct.py [hunk 1]
```diff
@@ -1,24 +1,25 @@
...
-#  GNU General Public License for more details.
+#SickRage is distributed in the hope that it will be useful,
+#but WITHOUT ANY WARRANTY; without even the implied warranty of
+#MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#GNU General Public License for more details.
 #
-# You should have received a copy of the GNU General Public License
-# along with SickRage.  If not, see <http://www.gnu.org/licenses/>.
+#You should have received a copy of the GNU General Public License
+#along with SickRage.  If not, see <http://www.gnu.org/licenses/>.
```
Comment (miigotu) [inline sickbeard/providers/newpct.py]: a space goes after comments `# sometext` not `#sometext`

File: sickbeard/providers/newpct.py [hunk 2]
```diff
@@ -71,6 +72,10 @@ def __init__(self):
             'q': ''
         }
 
+
+    def isEnabled(self):
+        return self.enabled
+
```
Comment (miigotu) [inline sickbeard/providers/newpct.py]: this def is removed for a reason....

File: sickbeard/providers/newpct.py [hunk 3]
```diff
@@ -110,18 +115,24 @@ def _doSearch(self, search_strings, search_mode='eponly', epcount=0, age=0, epOb
...
+                                    title_raw = torrent_row.get('title')
                                     title = self._processTitle(title_raw)
-
-                                    item = title, download_url, size
-                                    logger.log(u"Found result: %s " % title, logger.DEBUG)
-
-                                    items[mode].append(item)
+                                    
+                                    #Search results don't return torrent files directly, it returns show sheets so we must parse showSheet to access torrent.
+                                    data2 = self.getURL(showSheetURL)
```
Comment (miigotu) [inline sickbeard/providers/newpct.py]: a second getURL for every result is terrible, and will be hella slow parsing results.

**Comment by CristianBB** [comment]
I agree about make a second geturl, however newpct page changes so results returns only links to showsheet of episode with torrent link so its necessary.
If there is no other option, we should disable this provider.

**Comment by miigotu** [comment]
@CristianBB please email me the html from the search results and the showsheet that matches one of the results. I will see if I can work around the second getURL.

**Comment by CristianBB** [comment]
Sent.
I tried to search a way to construct manually the torrent file link from search result page, but there is no any place where appears the "id" of the torrent on the first page so I cannot figure how to construct it.
I hope you find something that has escaped to me :))

**Comment by CristianBB** [comment]
@miigotu did you find something?
I've been thinking about it, to avoid make the second getUrl for every result, maybe overwritting downloadResult method on newpct provider is the best way to avoid this.
The idea is:
- Remove second GetUrl and only save the link to episode sheet on doSearch method. The other atributes as title, size, etc also will be saved.
- Create a new downloadResult method on newpct, in that method I'll do the second getUrl (to parse the real download link).

In that way, we only do 1 more getUrl. What do you think? I can make that changes in a moment, but as always I'll do what you say =P

**Comment by CristianBB** [comment]
HI @fernandog 
Then we should disable this provider? Actually it's giving an error trying to parse web page.

**Comment by fernandog** [comment]
@miigotu

**Comment by miigotu** [comment]
@CristianBB sorry, there is a little flux atm with the SR team, please give us some time to sort out our management issues.

**Comment by CristianBB** [comment]
Don't worry @miigotu , you have my mail so you can write me every time you need. I'll wait for an answer about this provider and about NameParser fix ;)

**Comment by CristianBB** [comment]
@miigotu Here you have the code overwriting downloadResult method

**Pull request closed without merging**